### ISSUE TITLE
Add support for npm workspaces (multiple JavaScript root packages)

### ISSUE THREAD

**Issue by kriswuollett** [microsoft/vscode#134748]
Currently in order for debugging breakpoints, IntelliSense, and other functionality to work for JavaScript/TypeScript, the project's `package.json` needs to be at the root of the workspace window, this is being discussed in #12463, but I believe support should be there for multiple TS/JS "roots" due to support for [workspaces that was added to npm v7](https://docs.npmjs.com/cli/v7/using-npm/workspaces).  A request workspace support in general was requested in #116675, but closed due to inactivity.

Currently, developers need to change the window's root directory when working on a specific JS/TS project which may also imply needing to create duplicate `.vscode` settings folder for each package's folder. 

Using [Multi-root workspaces](https://code.visualstudio.com/docs/editor/workspaces#_multiroot-workspaces) seems like it is a workaround, but ideally the editor should know based on the context of the active editor or the `cwd` of a launch configuration what "root" for JS/TS should be used. Also using Multi-root workspaces would just create a long flattened list of projects in the Explorer Workspace window when ideally it should be organized hierarchically like projects in a monorepo.

**Comment by kriswuollett** [comment]
Here is an example repo to clone: kriswuollett/monorepo@fe846e76a4bb0e00532dcc871726bed21626a9b8 (initial commit on `microsoft-vscode-134748` branch there). I think the issue may be more of figuring out what settings I may be missing like @IllusionMH pointing out to try project references, which I will try next.

My goal is to debug & develop a provider in [NextAuth.js](https://next-auth.js.org/) in the context of my own app. In some method I wish to "vendor in" a repo with `git submodule` and use a local file path instead of a version in `packages.json` for it. Originally I was using Nx, but figured I should try to use as plain of a setup as possible when I couldn't get debugging working with it, see nrwl/nx#4154. Just want to make sure I can debug properly first before I add in the complexity of a vendored dependency.

There is a `README.md` file the example repo with some repro instructions. When running from the root directory, the breakpoint gets set on some processed file instead of the original. There also is a `/api/throw` endpoint to try exception breakpoints.

<img width="1617" alt="Screen Shot 2021-10-11 at 3 48 22 PM" src="https://user-images.githubusercontent.com/1834192/136850200-15ca48cb-433e-4499-ae56-03947379dc10.png">

**Comment by kriswuollett** [comment]
Made some progress. I may close this issue myself as works as intended once I try adding local npm dependency to the test project.

Using the `api/throw` endpoint came in handy. I tried again to just use `cwd`/`localRoot` as `"${workspaceFolder}/src/js"` and set up project references in a `tsconfig.json` at `src/js` with these contents:

```json
{
  "references": [{ "path": "./packages/batteries-included" }],
  "files": [],
  "include": [],
  "exclude": ["**/node_modules"],
  "compilerOptions": {
    "allowJs": true
  }
}
```
 
Launched the server, then enabled 'Caught Exceptions' and finally hit that api. The code stops in a `throw.ts` file somewhere. When you set a breakpoint in there it reveals where it thinks the file is at in the Breakpoints panel:

<img width="1923" alt="source_map_error" src="https://user-images.githubusercontent.com/1834192/136877642-4205a281-0513-4407-a1be-de2bd30b4026.png">

That path didn't look valid so I added this `sourceMapPathOverrides`: `"webpack://?:*/*": "${workspaceFolder}/src/js/packages/batteries-included/*"`, and now the breakpoints work as expected:

<img width="2672" alt="source_map_fixed_it" src="https://user-images.githubusercontent.com/1834192/136877700-995e0623-e2c8-4763-b857-af4bc58bc49f.png">

Not sure why adding `-internal` to the override wasn't needed?  

Removing the `src/js/tsconfig.json` file didn't change anything because I believe [IntelliSense does know about multiple JavaScript projects](https://code.visualstudio.com/docs/nodejs/working-with-javascript#_javascript-projects-jsconfigjson) by using the presence of `jsconfig.json` files. If it is a TypeScript project, then [`tsconfig.json` can be used instead](https://code.visualstudio.com/docs/nodejs/working-with-javascript#_migrating-to-typescript). I also realized that source in `.../__tests__` were JavaScript files that weren't referred to in `tsconfig.json`'s `include`, so I just gave them a `tsx` extension. So that fixed code navigation in things like `import Home from '@pages/index'` as well.

Example code now updated: kriswuollett/monorepo@7ba2786
